packages/cloud-cognitive/src/components/Datagrid/Datagrid.test.js

```diff
@@ -133,4 +733,471 @@ describe(componentName, () => {
       }
     );
   });
+
+  //Empty State
+  it('renders an empty table', () => {
+    render(<EmptyUsage data-testid={dataTestId}></EmptyUsage>);
+    expect(
+      screen.getByRole('table').getElementsByTagName('tbody')[0].className
+    ).toEqual('c4p--datagrid__empty-state-body');
+  });
+
+  /*
+  it('Initial Load', () => {
+
+    render(<InitialLoad data-testid={dataTestId}></InitialLoad>);
+  });*/
+
+  //TODO: Ask Mark, what should we check / expect this to be?
+  it('Infinite Scroll', () => {
+    render(<InfiniteScroll data-testid={dataTestId}></InfiniteScroll>);
+
+    expect(
+      screen
+        .getByRole('table')
+        .getElementsByTagName('tbody')[0]
+        .getElementsByTagName('div')[0].classList[0]
+    ).toBe('c4p--datagrid__virtual-scrollbar');
+
+    /*const scrollContainer = screen.queryAllByRole('rowgroup')[0];
+    
+    scrollContainer.addEventListener('scroll');
+
+    fireEvent.scroll(scrollContainer, {target: {y: 100}});*/
+  });
+
+  //Ten Thousand Entries
+  it('renders Ten Thousand table entries', () => {
```

Comment:
You verified the table height which is great. It would be great if you could also verify the number of rows in the table (which should be 10,000 in this case).